Add new Use of Funds modal to all banners

Abban opened the pull request:
- Fixed styles to work with desktop banners
- Added dark mode swatch and skins
- Fixed flows in the mobile banner flows where we have multiple modals open
- Added specific banner feature tests for these flows
- Finally started making banners accessible by adding a readable text to close icons 🎉 
- Fixed issue in locale factory for WPDE

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add new Use of Funds modal to Desktop banners

- Fix styles for desktop
- Add missing colours

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add new Use of Funds modal to EN Desktop banners

- Add dark mode to new use of funds

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Fix up new Use of Funds on Mobile DE banners

- Remove unused infographic component
- Tweak styles to make full width work better
- Make sure modalOpened and modalClosed only fire once per modal
- Add UoF tests, and separate mobile and desktop

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Fix up new Use of Funds on Mobile EN banners

- Improve scrolling functionality and tests
- Fix tests for clos events in FillPageBanner and BannerConductor
- Add minerva skin to uof swatch
- Add missing modal opened and closed events

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add new Use of Funds to iPad banners

- Adds new UoF and tests
- Sets form back to fixed width

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add new Use of Funds to WPDE desktop banners

- Adds new UoF and tests
- Fixes LocaleFactory

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add new Use of Funds to WPDE mobile banners

- Adds new UoF and tests

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add is-sr-only class

This allows us to put text for screen readers alongside
button icons.

Added close text to the messages and hooked up the main and
UoF close buttons.

Ticket: https://phabricator.wikimedia.org/T381956

moiikana commented on src/components/UseOfFunds2024/styles/swatches/color_dark.scss:
```diff
@@ -0,0 +1,43 @@
+	--benefits-list-icon: #{$white};
+	--benefits-background: #{$gray500};
+
+	--budget-line-background: #cecece;
```
I understood for consistency we would want to pull those 4 colors up to the beginning of the file and give them a name, right?

Abban replied:
Oh, the line background one should be but I've been treating the accordion colours as bespoke (eg outside our design system colours and individual per item). We did the same with the old UoF accordion and we do it with the payment logos.

gbirke commented on src/components/ButtonClose/messages/ButtonClose.de.ts:
```diff
@@ -0,0 +1,7 @@
+import { TranslationMessages } from '@src/Translator';
+
+const translations: TranslationMessages = {
+	'close': 'Schließen Sie'
```
```suggestion
	'close': 'Schließen'
```

gbirke commented on src/components/UseOfFunds2024/styles/UseOfFunds.scss:
```diff
@@ -25,7 +46,15 @@
+
+		p.use-of-funds-summary-text {
+			line-height: 1.8;
+			font-size: 18px;
```
I think this should be 16px for mobile, it looks very large on small screens and different than the rest of the UoF text

Abban replied:
Oh it was a typo

gbirke reviewed (approved):
On Firefox Mac (Desktop EN and DE and iPad), when you open the UoF banner, the close button is focused, giving the circular button an unsightly blue square outline. It does not happen every time, but when you load the banner and immediately show the UoF, it's immediately visible

moiikana commented:
> On Firefox Mac (Desktop EN and DE and iPad), when you open the UoF banner, the close button is focused, giving the circular button an unsightly blue square outline. It does not happen every time, but when you load the banner and immediately show the UoF, it's immediately visible

same behaviour on my linux firefox

Abban committed:
Fix PR issues

- Decrease font size on small screens
- Fix focus on modal open
- Fix German close message
- Move accordion line break colour into palette

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Add comments to external colours in swatches

We use some external branding colours in our swatches. These
are set to css variables using the hex codes directly rather
than being put into our palettes. As this is confusing I added
explanation of why to each of the places they're being defined.

Also removes all usages of palette variables in these var
assignments to make sure the external colours are completely
separated from our own.

Ticket: https://phabricator.wikimedia.org/T381956

Abban committed:
Update content
